### PR TITLE
chore: bump operator-rs 0.60.1

### PR DESCRIPTION
**Pull request by NickLarsenNZ**
# Description

- Use new label builders.

> [!NOTE]
> I have no idea where the _defintion_ spelling mistake is coming in, but it is not from this repo.

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] CRD documentation for all fields, following the [style guide](https://docs.stackable.tech/home/nightly/contributor/style-guide).
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
- [ ] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated. Follows the [style guide](https://docs.stackable.tech/home/nightly/contributor/style-guide).
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

### CODE REVIEW

File: rust/operator-binary/src/controller.rs [hunk 1]
```diff
@@ -20,12 +20,15 @@ use product_config::{
...
-    PodSecurityContextBuilder, SecretOperatorVolumeSourceBuilder, VolumeBuilder,
+    ObjectMetaBuilderError, PodSecurityContextBuilder, SecretOperatorVolumeSourceBuilder,
+    SecretOperatorVolumeSourceBuilderError, VolumeBuilder,
 };
 use stackable_operator::client::GetApi;
 use stackable_operator::commons::authentication::tls::{CaCert, TlsVerification};
 use stackable_operator::commons::s3::S3ConnectionSpec;
+use stackable_operator::commons::secret_class::SecretClassVolumeError;
 use stackable_operator::k8s_openapi::api::core::v1::SecretVolumeSource;
+use stackable_operator::kvp::{LabelError, Labels};
```
Comment (maltesander) [inline rust/operator-binary/src/controller.rs:31]: Can we summarize these imports?

**Review by maltesander — COMMENTED** [review]
Just an import thing, looks good otherwise!

File: deploy/helm/edc-operator/crds/crds.yaml [hunk 2]
```diff
@@ -32,21 +32,22 @@ spec:
                     ionos:
                       properties:
                         s3:
-                          description: Operators are expected to define fields for this type in order to work with S3 buckets.
+                          description: An S3 bucket definition, it can either be a reference to an explicit S3Bucket object, or it can be an inline defintion of a bucket. Read the [S3 resources concept documentation](https://docs.stackable.tech/home/nightly/concepts/s3) to learn more.
```
Comment (maltesander) [inline deploy/helm/edc-operator/crds/crds.yaml:35]: Damn :D

File: rust/operator-binary/src/controller.rs (hunk 1 above)
Comment (NickLarsenNZ) [reply rust/operator-binary/src/controller.rs:31]: Do you mean like with `::*`?

I guess that raises the debate of explicit vs implicit imports.

Perhaps for the stuff used in every operator, we can make a prelude module in operator-rs which re-exports stuff.

Comment (Techassi) [reply rust/operator-binary/src/controller.rs:31]: Use a grouped import is I guess what he meant. So like `use foo::{a, b, c d::{e, f}}`.

File: deploy/helm/edc-operator/crds/crds.yaml (hunk 2 above)
Comment (Techassi) [reply deploy/helm/edc-operator/crds/crds.yaml:35]: As this not critical at all, I would say keep this PR as is and incorporate the fix for the typo in a new bump PR in the future. I will open a PR for `operator-rs`.

Comment (Techassi) [reply deploy/helm/edc-operator/crds/crds.yaml:35]: Fix-up PR: https://github.com/stackabletech/operator-rs/pull/710

File: rust/operator-binary/src/controller.rs [hunk 3]
```diff
@@ -785,8 +833,13 @@ fn add_s3_volume_and_volume_mounts(
 ) -> Result<()> {
     if let Some(s3_conn) = s3_conn {
         if let Some(credentials) = &s3_conn.credentials {
-            pb.add_volume(credentials.to_volume("s3-credentials"));
-            cb_druid.add_volume_mount("s3-credentials", STACKABLE_SECRETS_DIR);
+            let volume_name = "s3-credentials";
```
Comment (Techassi) [inline rust/operator-binary/src/controller.rs:836]: We might even consider to move this into a `const`. What is your opinion on that?

**Review by Techassi — COMMENTED** [review]
Very clean PR, just one small idea.

File: rust/operator-binary/src/controller.rs (hunk 1 above)
Comment (maltesander) [reply rust/operator-binary/src/controller.rs:31]: yes @Techassi

File: rust/operator-binary/src/controller.rs (hunk 3 above)
Comment (NickLarsenNZ) [reply rust/operator-binary/src/controller.rs:836]: haha, I had that exact thought at the time of writing. But thought it's a negligible performance improvement (and technically negligible size increase :rofl:).

Given it has been mentioned, I'll do it :smile:

File: rust/operator-binary/src/controller.rs (hunk 1 above)
Comment (NickLarsenNZ) [reply rust/operator-binary/src/controller.rs:31]: Can do :+1:

File: rust/operator-binary/src/controller.rs (hunk 3 above)
Comment (NickLarsenNZ) [reply rust/operator-binary/src/controller.rs:836]: It does mean that the error context now looks like:

```rust
                CredentialsToVolumeSnafu {
                    volume_name: VOLUME_NAME,
                },
```

File: rust/operator-binary/src/controller.rs (hunk 1 above)
Comment (NickLarsenNZ) [reply rust/operator-binary/src/controller.rs]: Resolved in dca6a44ffdf8951892e9921a5aeb5b97829bb1b9

File: rust/operator-binary/src/controller.rs (hunk 3 above)
Comment (NickLarsenNZ) [reply rust/operator-binary/src/controller.rs]: Resolved in a8b2bc349f8cf5ab02e7dd8e16de17f6c888e63c

File: rust/operator-binary/src/controller.rs [hunk 4]
```diff
@@ -785,8 +834,13 @@ fn add_s3_volume_and_volume_mounts(
 ) -> Result<()> {
     if let Some(s3_conn) = s3_conn {
         if let Some(credentials) = &s3_conn.credentials {
-            pb.add_volume(credentials.to_volume("s3-credentials"));
-            cb_druid.add_volume_mount("s3-credentials", STACKABLE_SECRETS_DIR);
+            const VOLUME_NAME: &'static str = "s3-credentials";
```
[the remark this answers is not in the document]
Comment (NickLarsenNZ) [reply rust/operator-binary/src/controller.rs]: Resolved in b748c36e045032c293cdef888ada51316a8142dd

File: rust/operator-binary/src/controller.rs [hunk 5]
```diff
@@ -785,8 +834,13 @@
...
```
[the remark this answers is not in the document]
Comment (NickLarsenNZ) [reply rust/operator-binary/src/controller.rs]: Resolved in b748c36e045032c293cdef888ada51316a8142dd

**Review by Techassi — APPROVED** [review]

**Pull request merged**